monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -545,28 +543,38 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
-    @staticmethod
-    def is_latest_report_exists():
+    @classmethod
+    def get_latest_event_timestamp(cls) -> float:
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
+
+        # TODO: Add `get_latest_event` to the IAgentEventRepository
+        agent_events = cls._agent_event_repository.get_events()
+        latest_timestamp = max(agent_events, key=lambda event: event.timestamp).timestamp
+
+        return latest_timestamp
+
+    @classmethod
+    def is_latest_report_exists(cls):
```

Comment:
`latest_report_exists` is already a binary expression, no need to prepend `is_`

Reply:
but a better name would be `is_report_outdated`, because it doesn't check if it exists, it checks if it's up to date

Reply:
Better yet, "report_is_outdated()". Then your condition says `if report_is_outdated()`, which is readable as proper English.